lib/kitchen-sink/map.ex

```diff
@@ -138,4 +138,40 @@ defmodule KitchenSink.Map do
   def rename_key(_,_,_) do
     %{}
   end
+
+  def new_nested_map(key_list, value) do
+    key_list
+    |> Enum.reverse
+    |> Enum.reduce(value, fn (key, acc) -> %{key => acc} end)
+    |> Map.new
+  end
+
+  defp renamed_key(map) do
+    fn
+      ({old_key, [root_key | key_list]}) ->
+        value = Map.get(map, old_key)
+        {root_key, new_nested_map(key_list, value)}
+
+      ({old_key, new_key}) ->
+        value = Map.get(map, old_key)
+        {new_key, value}
+    end
+  end
+
+  @doc """
+  renames many keys in a map
```

Comment:
Need more detailed doc. Probably good to use doctest here too for transformational functions.